src/collectors/windows.plugin/perflib-memory.c

```diff
@@ -7,8 +7,155 @@
+    RRDSET *pages;
+    RRDDIM *rd_page_read;
+    RRDDIM *rd_page_write;
+
+    COUNTER_DATA pageReadsTotal;
+    COUNTER_DATA pageWritesTotal;
+    COUNTER_DATA pageInputTotal;
+    COUNTER_DATA pageOutputTotal;
+};
+
+struct system_pool {
+    RRDSET *pool;
+    RRDDIM *rd_paged;
+    RRDDIM *rd_nonpaged;
+
+    COUNTER_DATA pagedData;
+    COUNTER_DATA nonPagedData;
+};
+
+struct swap localSwap = { 0 };
+struct system_pool localPool = { 0 };
+
+void initialize_swap_keys(struct swap *p) {
+    // SWAP Operations
+    p->pageReadsTotal.key = "Page Reads/sec";
+    p->pageWritesTotal.key = "Page Writes/s";
+
+    // Swap Pages
+    p->pageInputTotal.key = "Pages Input/sec";
+    p->pageOutputTotal.key = "Pages Output/s";
+}
+
+void initialize_pool_keys(struct system_pool *p) {
+    p->pagedData.key = "Pool Paged Bytes";
+    p->nonPagedData.key = "Pool Nonpaged Bytes";
+}
+
 static void initialize(void) {
-    ;
+    initialize_swap_keys(&localSwap);
+    initialize_pool_keys(&localPool);
+}
+
+static void do_memory_swap(PERF_DATA_BLOCK *pDataBlock, PERF_OBJECT_TYPE *pObjectType, int update_every)
+{
+    perflibGetObjectCounter(pDataBlock, pObjectType, &localSwap.pageReadsTotal);
+    perflibGetObjectCounter(pDataBlock, pObjectType, &localSwap.pageWritesTotal);
+    perflibGetObjectCounter(pDataBlock, pObjectType, &localSwap.pageInputTotal);
+    perflibGetObjectCounter(pDataBlock, pObjectType, &localSwap.pageOutputTotal);
+
+    if (!localSwap.operations) {
+        localSwap.operations = rrdset_create_localhost(
+            "mem"
+        , "swap_operations", NULL
+        , "swap"
+        , "mem.swap_operations"
+        , "Swap Operations"
+        , "operations/s"
+        , PLUGIN_WINDOWS_NAME
+        , "PerflibMemory"
+        , NETDATA_CHART_PRIO_MEM_SWAPIO
+        , update_every
+        , RRDSET_TYPE_STACKED
+        );
+
+        localSwap.rd_op_read = rrddim_add(localSwap.operations, "read", NULL,
+                                          1, 1, RRD_ALGORITHM_INCREMENTAL);
+        localSwap.rd_op_write = rrddim_add(localSwap.operations, "write", NULL,
+                                           1, -1, RRD_ALGORITHM_INCREMENTAL);
+    }
+
+    rrddim_set_by_pointer(localSwap.operations,
+                          localSwap.rd_op_read,
+                          (collected_number)localSwap.pageReadsTotal.current.Data);
+
+    rrddim_set_by_pointer(localSwap.operations,
+                          localSwap.rd_op_write,
+                          (collected_number)localSwap.pageWritesTotal.current.Data);
+    rrdset_done(localSwap.operations);
+
+    if (!localSwap.pages) {
+        localSwap.pages = rrdset_create_localhost(
+            "mem"
+        , "swap_pages", NULL
+        , "swap"
+        , "mem.swap_pages"
+        , "Swap Pages"
+        , "pages/s"
+        , PLUGIN_WINDOWS_NAME
+        , "PerflibMemory"
+        , NETDATA_CHART_PRIO_MEM_SWAP_PAGES
+        , update_every
+        , RRDSET_TYPE_STACKED
+        );

```

Comment:
Why do you think we need pages and not bytes? We know the page size, right?

Reply:
This was defined weeks ago in previous discussions we had in our daily meeting. Let me remember you: :)

- We are converting previously merged codes in https://github.com/netdata/netdata/blob/cf768b84abb0e0da521df466217843c40cec9908/src/go/plugin/go.d/modules/windows/charts.go#L536-L547 , for users have conditions to compare netdata collecting data directly from Windows, and `go.d.plugin` collecting with `windows exporter`. Why did you merge pages when you could also convert to bytes? :thinking: 
- We already had a chart named `mem.swap` showing bytes, so, it does not make sense to have another.

Any additional doubt, please, let me know! 
Thank you for your help!

Reply:
My doubt is that showing pages is a useless chart

Reply:
> We can always add it 

Why add? Can you explain why you decided to add them in the first place?

We don't show disk space in blocks but in bytes. Why show swap IO in pages?

Reply:
> > We can always add it
> 
> Why add? Can you explain why you decided to add them in the first place?
> 
> We don't show disk space in blocks but in bytes. Why show swap IO in pages?

For example windows resource monitor has "Hard Faults/Sec" which is hard **page** faults per sec.

Reply:
I think you use `Page Faults/sec` for it.

Reply:
`Page Faults/sec` is both hard and soft faults.

> Pages Input/sec: Pages Input/sec is the total number of pages read from disk to resolve hard page faults

If hard page faults result in reading a single page and it is always one-to-one then it is ok to have this metric but it should be called hard faults.

---

> Pages Output/sec: Shows the rate at which pages are written to disk to free up space in physical memory. A high rate of pages output might indicate a memory shortage. 

This doesn't look useful at all.

Reply:
Come on,  lets add it and if it is not useful at all (we can get feedback anyway) we will improve/drop it.

Reply:
I am not sure why discussions should be after and not before. I am of the opposite opinion. You can add it of course if you want.